torch_geometric/sampler/neighbor_sampler.py

```diff
@@ -124,16 +125,33 @@ def __init__(
             # TODO support `FeatureStore` with no edge types (e.g. `Data`)
             feature_store, graph_store = data
 
+            # Obtain all node and edge metadata:
+            node_attrs = feature_store.get_all_tensor_attrs()
+            edge_attrs = graph_store.get_all_edge_attrs()
+
             # TODO support `collect` on `FeatureStore`:
             self.node_time_dict = None
             if time_attr is not None:
+                # If the `time_attr` is present, we expect that `GraphStore`
```

Comment:
cc @mananshah99. I currently cannot think of a more elegant way to support this. IMO, we need to extend `GraphStore` to support some form node-level metadata as well.

Reply:
Hm, so it seems like the requirement here is that GraphStore should be able to return `csc()`, sorted by column, and sorted by time within local neighborhoods. Why don't we just extend the layout conversion functionality here to support `GraphStore.csc(..., src_node_time: Optional[torch.Tensor] = None)`, since this all happens in-memory anyway? We can handle this in `_edge_to_layout` there.

The alternative would be to allow GraphStore to support node-level metadata as you mentioned, and one argument in favor of this would be temporal sampling. I am also open to this; happy to discuss more in-person :)

Reply:
I have thought about additional an attribute to `csc` as well. It might work here indeed. What happens if we call `csc` with a time tensor though? Do we ignore it and simply trust that the data is formatted correctly?